src/components/Common/NumericInput.js

```diff
@@ -109,26 +118,71 @@ export class NumericInput extends Component {
     }
   }
 
+  zeroDecimals(decimals) {
+    if (decimals === '0' || decimals === 0 || !decimals) {
+      return true
+    } else return false
+  }
+
+  getTokensMinimumWithDecimals(minimum, decimals) {
+    if (this.zeroDecimals(decimals)) return
+
+    let minDecimals = new BigNumber(0)
+
+    minDecimals = minimum + 1 / Math.pow(10, decimals)
+
+    return minDecimals.toFixed(decimals)
+  }
+
+  getMinimumValue(dimension, decimals, minimum) {
+    if (dimension === 'tokens' && this.zeroDecimals(decimals)) {
+      return '1'
+    } else if (dimension === 'tokens') {
+      return this.getTokensMinimumWithDecimals(minimum, decimals)
+    } else if (dimension === 'percentage') {
+      return '0'
+    }
+  }
+
   render() {
     const { value, pristine, valid } = this.state
-    const { disabled, side, errorMessage, title, description, name } = this.props
+    const {
+      decimals,
+      description,
+      dimension,
+      disabled,
+      errorMessage,
+      extraClassName,
+      min,
+      name,
+      onClick,
+      placeholder,
+      title
+    } = this.props
+    const error = valid === INVALID ? <FormError errorMessage={errorMessage} /> : ''
+    const minimumValue = this.getMinimumValue(dimension, decimals, min)
+    const renderedValue = +value <= +minimumValue ? minimumValue : value
 
     return (
-      <InputField
-        disabled={disabled}
-        side={side}
-        name={name}
-        type="number"
-        errorMessage={errorMessage}
-        value={value}
-        pristine={pristine}
-        valid={valid}
-        title={title}
-        onKeyPress={this.onKeyPress}
-        onChange={this.onChange}
-        onPaste={this.onPaste}
-        description={description}
-      />
+      <div className={`sw-NumericInput ${extraClassName ? extraClassName : ''}`}>
+        <FormControlTitle title={title} description={description} />
+        <div className="sw-NumericInput_InputAndButtonContainer">
+          <TextField
+            disabled={disabled}
+            id={name}
+            min={minimumValue}
+            onChange={this.onChange}
+            onKeyPress={this.onKeyPress}
+            onPaste={this.onPaste}
+            placeholder={placeholder}
+            step="1"
+            type="number"
+            value={pristine ? '' : renderedValue}
+          />
+          <div onClick={onClick} className="sw-NumericInput_ButtonPlus" />
```

Comment:
@gabitoesmiapodo Is there a way to move the `+` button only outside of this component and add it in the parent component (`ReservedTokensInputBlock`?

I tried to encapsulate everything that's not the `NumericInput` itself into another component (something like `ReservedTokenNumericInput`... but I found myself dealing with the nuances of the field validation.

I don't think its necessary to do it in this PR, and will not affect the overall functionality as this component is only used in `ReservedTokensInputBlock`. But if it can be done, consider doing it in a future PR.